server/src/services/user/service.js

```diff
@@ -0,0 +1,65 @@
+import User from '../../models/user';
+import { compareHashData, hashData } from '../../utils/hashing';
+
+const userExists = async (email) => {
+  const query = {
+    email,
+  };
+  const userCountForGivenEmail = await User.countDocuments(query);
+
+  if (userCountForGivenEmail === 0) {
+    return false;
+  }
+  return true;
+};
+
+const createUser = async (userProperties) => {
+  const hashedPassword = await hashData(userProperties.password);
+  Object.assign(userProperties, { password: hashedPassword });
+  const newUser = User(userProperties);
+  const newUserInstance = await newUser.save();
+  return newUserInstance.toObject();
+};
+
+const usernameExists = async (username) => {
+  const query = {
+    username,
+  };
+  const userCountForGivenUsername = await User.countDocuments(query);
```

Comment:
You can abstract this into a function which can take query as input and use that fn for both username exists and email exists

Reply:
Yes but I don't want any other area of my code to be dealing with the queries that are specific to the user. So I thought of abstracting the query logic in the service only.     
Is this a good way to go? :smiley:

Reply:
username and email, both belong to the user, right?

Reply:
Yes, we are trying to deal with the queries part only on the service files. So that the other parts of the application don't have to deal with building queries :smiley: